### PR TITLE
MongoDB ODM 2.3 blog post

### PR DESCRIPTION
**Pull request by IonBazan**

### CODE REVIEW

File: source/blog/2021-12-01-mongodb-odm-2.3.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
+---
+title: "New Release: Doctrine MongoDB ODM 2.3 with Attributes, JSON Schema Validation and more"
```
Comment (malarzm) [inline source/blog/2021-12-01-mongodb-odm-2.3.md:2]:
```suggestion
title: "New Release: Doctrine MongoDB ODM 2.3 with Attributes, JSON Schema Validation, and more"
```

File: source/blog/2021-12-01-mongodb-odm-2.3.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+}
+```
+
+You may want to use [Rector](https://getrector.org/) with `DoctrineSetList::DOCTRINE_ODM_23` set
+to convert all your annotation mappings to attributes in seconds!
+
+## JSON Schema Validation
+
+MongoDB ≥ 3.6 offers the capability to validate documents during
+insertions and updates through a JSON schema associated to the collection.
```
Comment (malarzm) [inline source/blog/2021-12-01-mongodb-odm-2.3.md:48]:
```suggestion
insertions and updates through a JSON schema associated with the collection. 
```
but not sure about this one

File: source/blog/2021-12-01-mongodb-odm-2.3.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+to convert all your annotation mappings to attributes in seconds!
+
+## JSON Schema Validation
+
+MongoDB ≥ 3.6 offers the capability to validate documents during
+insertions and updates through a JSON schema associated to the collection. 
+[See MongoDB documentation](https://docs.mongodb.com/manual/core/schema-validation/).
+
+Doctrine MongoDB ODM now provides a way to take advantage of this functionality
+thanks to the new `#[Validation]` mapping.
```
Comment (malarzm) [inline source/blog/2021-12-01-mongodb-odm-2.3.md:52]: Given our "annotation reference" page is still not showing up in the menu I'd link to https://www.doctrine-project.org/projects/doctrine-mongodb-odm/en/latest/reference/annotations-reference.html#validation directly

File: source/blog/2021-12-01-mongodb-odm-2.3.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+
+```
+
+Once defined, those options will be added to the collection after running
+the ``odm:schema:create`` or ``odm:schema:update`` command.
+
+## Psalmified APIs
+
+Improved the documentation to make sure static analysis tools and IDEs know
```
Comment (malarzm) [inline source/blog/2021-12-01-mongodb-odm-2.3.md:93]:
```suggestion
In-code documentation has been immensely improved to make sure static analysis tools and IDEs know
```

File: source/blog/2021-12-01-mongodb-odm-2.3.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+about the right document classes returned from `DocumentManager`,
+`DocumentManager`, `ClassMetadata` and other public ODM APIs. This includes generics support
```
Comment (malarzm) [inline source/blog/2021-12-01-mongodb-odm-2.3.md:95]:
```suggestion
`ClassMetadata`, and other public APIs. This includes generics support
```
- DocumentManager was listed twice
- Oxford comma

File: source/blog/2021-12-01-mongodb-odm-2.3.md [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+when you extend `DocumentRepository`.
```
Comment (malarzm) [inline source/blog/2021-12-01-mongodb-odm-2.3.md:96]:
```suggestion
for your own repositories extending `DocumentRepository`.
```

File: source/blog/2021-12-01-mongodb-odm-2.3.md [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+## Deprecations
+
+Doctrine MongoDB ODM 2.3 introduces several minor deprecations:
+
+- The `Doctrine\ODM\MongoDB\Proxy\Resolver\ClassNameResolver` interface has been deprecated in favor 
+  of the `Doctrine\Persistence\Mapping\ProxyClassNameResolver` interface
+- Annotation classes no longer extend `Doctrine\Common\Annotations\Annotation` class
+- Annotation arguments switched to `@NamedArgumentConstructor` for Attribute compatibility
+- `@Inheritance` annotation has been removed as it was never used
+- Document Namespace Aliases (`'MyBundle:User`) - use fully qualified class names instead (`User::class`)
```
Comment (malarzm) [inline source/blog/2021-12-01-mongodb-odm-2.3.md:120]: I'd add a link to https://github.com/doctrine/mongodb-odm/blob/2.3.x/UPGRADE-2.3.md

**Comment by IonBazan** [comment]
@malarzm thanks for your review. I've fixed the issues you mentioned and updated the post date to 4 Dec. Feel free to change that according to the actual publish date.

**Review by malarzm — APPROVED** [review]

**Pull request merged**